test/org/zaproxy/zap/extension/pscanrulesBeta/ServletParameterPollutionScannerUnitTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+import static org.junit.Assert.assertEquals;
+
+public class ServletParameterPollutionScannerUnitTest extends PassiveScannerTest {
+
+    public static final String URI = "http://www.example.com/test/";
+
+    @Override
+    protected ServletParameterPollutionScanner createScanner() {
+        return new ServletParameterPollutionScanner();
+    }
+
+    @Test
+    public void GivenNoForms_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("");
+        scanHttpResponseReceive(msg);
+        assertEquals(alertsRaised.size(), 0);
+    }
+
+    @Test
+    public void GivenFormWithActionParameter_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action='ActionMan'>");
+        scanHttpResponseReceive(msg);
+        assertEquals(alertsRaised.size(), 0);
+    }
+
+    @Test
+    public void GivenFormWithNoActionParameter_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form />");
+        scanHttpResponseReceive(msg);
+        assertEquals(alertsRaised.size(), 1);
+    }
+
+    @Test
+    public void GivenTwoFormsWithNoActionParameter_WhenScanHttpResponseReceive_ThenOnlyOneAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form /><form />");
+        scanHttpResponseReceive(msg);
+        assertEquals(alertsRaised.size(), 1);
+    }
+
+    @Test
+    public void GivenFormWithNoActionParameter_WhenScanHttpResponseReceive_ThenAlertPopulated() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form />");
+        scanHttpResponseReceive(msg);
+        Alert alert = alertsRaised.get(0);
```

Comment:
It would be better if it asserted that the alert was raised, it's more clear an assertion error than `IndexOutOfBoundsException`.